files/scripts/graphdb.sh

```diff
@@ -98,4 +105,87 @@ function createRepositoryFromFile {
   fi
 }
 
+function interpolate {
+  local input=""
+  while IFS='' read -r line || [ -n "$line" ]; do
+    input="$input$line"$'\n'
+  done
+
+  vars=$(echo "${input}" | grep -o "\${[^}]*}" | sed 's/[${}]//g' | uniq)
+
+  local output="$input"
+  for var in $vars; do
+    value=${!var:-''}
+    if [ -n "$value" ]; then
+      output=${output//\$\{$var\}/$value}
+    fi
+  done
+  echo "${output}"
+}
+
+function cloudBackup {
+  BACKUP_TIMESTAMP="$(date +'%Y-%m-%d_%H-%M-%S')"
+  BACKUP_NAME="graphdb-backup-${BACKUP_TIMESTAMP}.tar"
+  export BACKUP_NAME BACKUP_TIMESTAMP
+
+  local backup_options=
+  backup_options=$(interpolate < "$1")
+
+  log "Creating cloud backup ${BACKUP_NAME}"
+
+  local response=
+  local response_status
+  response=$(mktemp)
+  response_status=$(curl -X POST \
+    -isSL \
+    -o "${response}" \
+    -w "Status=%{response_code}" \
+    --header "Authorization: Basic ${GRAPHDB_AUTH_TOKEN}" \
+    --header 'Content-Type: application/json' \
+    --header 'Accept: application/json' \
+    --data-binary "${backup_options}" \
+    --url "http://${GRAPHDB_SERVICE_NAME}:${GRAPHDB_SERVICE_PORT}/rest/recovery/cloud-backup")
+
+  if echo "${response_status}" | grep -q 'Status=200' ; then
```

Comment:
I've always hated if/elses when they can be avoided - can we invert the check and `return 1` or `exit 1` instead?